pkg/client/proxy_snapshot.go

```diff
@@ -46,26 +51,30 @@ func (c *ProxyClient) VolumeSnapshot(serviceAddress, volumeName string, labels m
 	}
 	recv, err := c.service.VolumeSnapshot(c.ctx, req)
 	if err != nil {
-		return "", errors.Wrapf(err, "failed to snapshot volume via proxy %v to %v", c.ServiceURL, serviceAddress)
+		return "", err
 	}
 
 	return recv.Snapshot.Name, nil
 }
 
 func (c *ProxyClient) SnapshotList(serviceAddress string) (snapshotDiskInfo map[string]*etypes.DiskInfo, err error) {
-	if serviceAddress == "" {
-		return nil, errors.Wrapf(ErrParameter, "failed to list snapshots")
+	input := map[string]string{
+		"serviceAddress": serviceAddress,
+	}
+	if err := checkParameters(input); err != nil {
+		return nil, errors.Wrap(err, "failed to list replicas for volume")
```

Comment:
"failed to list snapshots for volume" (Please recheck all logs to avoid any wrong message)

Reply:
Fixed.